db/compaction/compaction_picker.cc

```diff
@@ -878,21 +880,21 @@ namespace {
 // Test whether two files have overlapping key-ranges.
 bool HaveOverlappingKeyRanges(const Comparator* c, const SstFileMetaData& a,
                               const SstFileMetaData& b) {
-  if (c->Compare(a.smallestkey, b.smallestkey) >= 0) {
-    if (c->Compare(a.smallestkey, b.largestkey) <= 0) {
+  if (c->CompareWithoutTimestamp(a.smallestkey, b.smallestkey) >= 0) {
```

Comment:
Rest of changes to this file covered by `TimestampCompatibleCompactionTest.CompactFilesRangeCheckL0` and `TimestampCompatibleCompactionTest.CompactFilesRangeCheckL1`.